cli/client.go

```diff
@@ -2041,9 +2065,164 @@ func (conf *Config) checkUpdate(c *cli.Context) error {
+			"New CLI releases happen weekly; consider updating%s. Run 'viam update' or see https://docs.viam.com/cli/#install", updateInstructions)
+	}
+	return nil
+}
+
+// UpdateCLIAction updates the CLI to the latest version.
+func UpdateCLIAction(c *cli.Context, args emptyArgs) error {
+	// 1. check CLI to see if update needed, if this fails then try update anyways
+	latestVersion, latestVersionErr := latestVersion()
+	if latestVersionErr != nil {
+		warningf(c.App.ErrWriter, "CLI Update Check: failed to get latest release information: %w", latestVersionErr)
+	}
+	localVersion, localVersionErr := localVersion()
+	if localVersionErr != nil {
+		warningf(c.App.ErrWriter, "CLI Update Check: failed to get local release information: %w", localVersionErr)
+	}
+	if localVersion != nil && latestVersion != nil {
+		if localVersion.GreaterThanEqual(latestVersion) {
+			infof(c.App.Writer, "Your CLI is already up to date (version %s)", localVersion.Original())
+			return nil
+		}
+	}
+	// 2. check if cli managed by brew, if so attempt update. If it fails
+	// dont continue with binary replacement to avoid putting brew out of sync
+	managedByBrew, err := checkAndTryBrewUpdate()
+	if err != nil {
+		return errors.Errorf("CLI update failed: %v", err)
+	}
+	// try the binary replacement process because not managed by brew
+	if !managedByBrew {
+		// 3. get the local version binary path (use full path if no symlinks)
+		execPath, err := os.Executable()
+		if err != nil {
+			return errors.Errorf("CLI update failed: failed to get executable path: %v", err)
+		}
+		localBinaryPath, err := filepath.EvalSymlinks(execPath)
+		if err != nil {
+			localBinaryPath = execPath
+		}
+		directoryPath := filepath.Dir(localBinaryPath)
+
+		// 4. get the latest binary (from storage.googleapis.com) and write it into a temp file
+		binaryURL := binaryURL()
+		latestBinaryPath, err := downloadBinaryIntoDir(binaryURL, directoryPath)
+		defer os.Remove(latestBinaryPath) //nolint:errcheck
+		if err != nil {
+			return errors.Errorf("CLI update failed: failed to download binary: %v", err)
+		}
+
+		// 5. replace the old binary with the new one
+		if err := replaceBinary(localBinaryPath, latestBinaryPath); err != nil {
+			return errors.Errorf("CLI update failed: failed to replace binary: %v", err)
+		}
+	}
+	infof(c.App.Writer, "Your CLI has been successfully updated")
+	return nil
+}
+
+func checkAndTryBrewUpdate() (bool, error) {
+	if runtime.GOOS == "darwin" {
+		if _, err := exec.LookPath("brew"); err == nil {
+			// Check if viam is actually managed by brew
+			err := exec.Command("brew", "list", "viam").Run()
+			if err == nil {
+				// viam is managed by brew - try upgrade
+				out, err := exec.Command("brew", "upgrade", "viam").CombinedOutput()
+				if err == nil {
+					if strings.Contains(string(out), "already installed") {
+						// edge case: latest version released but brew has not updated yet
+						return false, errors.New("the latest version is not on brew yet")
```

Comment:
should this return `true` for `managedByBrew`? Otherwise we'd try the other binary replacement process at line 2097 which could create some weird interactions.